js/apps/system/_admin/aardvark/APP/react/src/views/collections/indices/addIndex/IndexFormField.tsx

```diff
@@ -1,66 +1,102 @@
   name: string;
   type: string;
+  options?: Array<OptionType>;
   tooltip?: string;
+  group?: string;
   isRequired?: boolean;
   isDisabled?: boolean;
 };
 
 export const IndexFormField = ({
   field,
+  render,
   index
 }: {
   field: IndexFormFieldProps;
-  index: number;
+  index?: number;
+  render?: (props: {
+    field: IndexFormFieldProps;
+    index?: number;
+    autoFocus: boolean;
+  }) => JSX.Element;
 }) => {
+  const autoFocus = index === 0;
+  if (render && field.type === "custom") {
+    return render({ field, index, autoFocus });
+  }
   switch (field.type) {
     case "boolean":
       return (
         <>
-          <FormLabel htmlFor={field.name}>{field.label}</FormLabel>
+          <FormLabel margin="0" htmlFor={field.name}>
+            {field.label}
+          </FormLabel>
           <SwitchControl
             isDisabled={field.isDisabled}
             isRequired={field.isRequired}
             name={field.name}
           />
-          {field.tooltip && <InfoTooltip label={field.tooltip} />}
+          {field.tooltip ? <InfoTooltip label={field.tooltip} /> : <Box></Box>}
         </>
       );
     case "number":
       return (
         <>
-          <FormLabel htmlFor={field.name}>{field.label}</FormLabel>
+          <FormLabel margin="0" htmlFor={field.name}>
+            {field.label}
+          </FormLabel>
           <InputControl
             isDisabled={field.isDisabled}
-            inputProps={{ type: "number", autoFocus: index === 0 }}
+            inputProps={{ type: "number", autoFocus }}
             isRequired={field.isRequired}
             name={field.name}
           />
-          {field.tooltip && <InfoTooltip label={field.tooltip} />}
+          {field.tooltip ? <InfoTooltip label={field.tooltip} /> : <Box></Box>}
+        </>
+      );
+    case "multiSelect":
+      return (
+        <>
+          <FormLabel margin="0" htmlFor={field.name}>
+            {field.label}
+          </FormLabel>
+          <MultiSelectControl
+            isDisabled={field.isDisabled}
+            selectProps={{
+              autoFocus,
+              options: field.options
+            }}
+            isRequired={field.isRequired}
+            name={field.name}
+          />
+          {field.tooltip ? <InfoTooltip label={field.tooltip} /> : <Box></Box>}
         </>
       );
     case "text":
     default:
       return (
         <>
-          <FormLabel htmlFor={field.name}>{field.label}</FormLabel>
+          <FormLabel margin="0" htmlFor={field.name}>
+            {field.label}
+          </FormLabel>
           <InputControl
             isDisabled={field.isDisabled}
             isRequired={field.isRequired}
             name={field.name}
-            inputProps={{ autoFocus: index === 0 }}
+            inputProps={{ autoFocus }}
           />
-          {field.tooltip && <InfoTooltip label={field.tooltip} />}
+          {field.tooltip ? <InfoTooltip label={field.tooltip} /> : <Box></Box>}
```

Comment:
This is nice. I think we can make this a practice. Recently started noticing issues with returning `null` 😛